client/quota.go

```diff
@@ -50,20 +50,18 @@ type QuotaValues struct {
 
 // EnsureQuota creates a quota group or updates an existing group.
 // The list of snaps can be empty.
-func (client *Client) EnsureQuota(groupName string, parent string, snaps []string, maxMemory quantity.Size) (changeID string, err error) {
+func (client *Client) EnsureQuota(groupName string, parent string, snaps []string, constraints *QuotaValues) (changeID string, err error) {
 	if groupName == "" {
 		return "", fmt.Errorf("cannot create or update quota group without a name")
 	}
 	// TODO: use naming.ValidateQuotaGroup()
 
 	data := &postQuotaData{
-		Action:    "ensure",
-		GroupName: groupName,
-		Parent:    parent,
-		Snaps:     snaps,
-		Constraints: &QuotaValues{
-			Memory: maxMemory,
-		},
+		Action:      "ensure",
+		GroupName:   groupName,
+		Parent:      parent,
+		Snaps:       snaps,
+		Constraints: constraints,
```

Comment:
Do we need to handle the case where constrains is nil? This wasn't possible with the old code.